agent/connect/ca/provider_consul.go

```diff
@@ -63,8 +69,7 @@ func (c *ConsulProvider) Configure(cfg ProviderConfig) error {
 		return err
 	}
 	c.config = config
-	hash := sha256.Sum256([]byte(fmt.Sprintf("%s,%s,%v", config.PrivateKey, config.RootCert, cfg.IsPrimary)))
-	c.id = connect.HexString(hash[:])
+	c.id = hexStringHash(fmt.Sprintf("%s,%s,%s,%d,%v", config.PrivateKey, config.RootCert, config.PrivateKeyType, config.PrivateKeyBits, cfg.IsPrimary))
```

Comment:
This `id` is used to key the builtin proxy configuration in the state store (much like a cache-key). Adding the key type/bits here allows it to have two builtin proxy roots defined simultaneously during the rotation.

Reply:
At first I thought `RootCert` should have already been unique here, but after reading the tests I realize now that this input struct is a `ConsulCAProviderConfig` , not the `CARoots`, which means in most cases when `PrivateKey{Type,Bits}` are set, `RootCert` would be the zero value.